### PR TITLE
Onboarding Intro - Pixels

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1207723831937959
CC: @SabrinaTardio 

**Description**:
Add Pixels to Onboarding Intro flow.
[Pixel Definitions](https://app.asana.com/0/1206329551987282/1207592597178194/f) `#2`, `#3`, `#4`.

Pixels `#2` `#3` are unique as per [comment](https://app.asana.com/0/0/1207592597178194/1207738121425440/f).

**Steps to test this PR**:

**SCENARIO 1 - New Intro Flow**
1. Launch the New Onboarding Intro Flow 
2. Wait for the Dax Dialog “Ready for a better internet?” appears
3. Ensure that `m.preonboarding.intro.shown` `Pixel #2` is fired.
4. Tap “Let’s do it”
5. Wait for the Browser Chart to appear
6. Ensure that `m.preonboarding.comparison.chart.shown` `Pixel #3` is fired.
7. Tap “Choose Your Browser”
8 Ensure that `m.preonboarding.choose.browser.pressed` `Pixel #4` is fired.  
9. Repeat the steps and ensure that `m.preonboarding.intro.shown` and `m.preonboarding.comparison.chart.shown` are not fired again

**SCENARIO 2 - Old Onboarding Flow**
1. Enable Old onboarding flow (Edit Scheme -> Arguments -> Environment Variables -> Onboarding -> enable)
2. Launch App
3. Ensure that the `m.preonboarding.intro.shown` `Pixes #2` is fired.
4. Repeat the steps and ensure that `m.preonboarding.intro.shown` is not fired again.

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/DaxOnboardingViewController.swift [hunk 1]
```diff
@@ -61,6 +73,8 @@ class DaxOnboardingViewController: UIViewController, Onboarding {
         daxDialogContainerHeight.constant = daxDialog?.calculateHeight() ?? 0
         button.displayDropShadow()
         daxIcon.isHidden = true
+
+        pixelReporting.trackOnboardingIntroImpression()
```
Comment (alessandroboron) [inline DuckDuckGo/DaxOnboardingViewController.swift:77]: My understanding is to send the new pixel for the old flow too.

File: DuckDuckGo/OnboardingExperiment/Pixels/OnboardingPixelReporterStore.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import Foundation
+import Core
+
+protocol OnboardingPixelReporterStorage: AnyObject {
+    var hasFiredIntroScreenShownPixel: Bool { get set }
+    var hasFiredComparisonChartShownPixel: Bool { get set }
+}
+
+final class OnboardingPixelReporterStore: OnboardingPixelReporterStorage {
```
Comment (SabrinaTardio) [inline DuckDuckGo/OnboardingExperiment/Pixels/OnboardingPixelReporterStore.swift]: I don’t think we need an explicit store… There is already a UniquePixel class that does everything for you already.
The only caveat is that we need to allow the pixel suffix to be _unique as well as _u.

Comment (alessandroboron) [reply DuckDuckGo/OnboardingExperiment/Pixels/OnboardingPixelReporterStore.swift]: Good catch, I will make the change!

File: DuckDuckGo/OnboardingExperiment/Pixels/OnboardingPixelReporter.swift [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+extension UniquePixel: OnboardingPixelFiring {
+    static func fire(pixel: Pixel.Event, withAdditionalParameters params: [String: String], includedParameters: [Pixel.QueryParameters]) {
+        self.fire(pixel: pixel, withAdditionalParameters: params, includedParameters: includedParameters, onComplete: { _ in })
+    }
+}
+
+// MARK: - OnboardingPixelReporter
+
+protocol OnboardingIntroImpressionReporting {
```
Comment (SabrinaTardio) [inline DuckDuckGo/OnboardingExperiment/Pixels/OnboardingPixelReporter.swift:43]: I would use a unique protocol

**Comment by brindy** [comment]
This LGTM but I'll let @SabrinaTardio approve it since she had feedback before.

**Review by SabrinaTardio — APPROVED** [review]
Works as expected!

**Pull request merged**